CoreScriptsRoot/CoreScripts/PerformanceStatsManagerScript.lua

```diff
@@ -0,0 +1,219 @@
+
+--[[
+		Filename: PerformanceStatsManagerScript.lua
+		Written by: dbanks
+		Description: Handles performance stats gui.
+--]]
+
+--[[ Services ]]--
+local PlayersService = game:GetService("Players")
+local Settings = UserSettings()
+local GameSettings = Settings.GameSettings
+local CoreGuiService = game:GetService('CoreGui')
+
+--[[ Convenience function]]--
+function WaitForFolder(pathPieces, parent)
+  if parent == nil then
+    parent = CoreGuiService
+  end
+  
+  if (table.getn(pathPieces) == 0) then 
+    return parent
+  else 
+    local first = table.remove(pathPieces, 1)
+    local newParent = parent:WaitForChild(first)
+    return WaitForFolder(pathPieces, newParent)
+  end
+  
+end
+
+--[[ Modules ]]--
+local folder = CoreGuiService:WaitForChild("RobloxGui")
+folder = folder:WaitForChild("Modules")
+folder = folder:WaitForChild("Stats")
+
+local AllStatsAggregatorsClass = require(folder:WaitForChild( 
+    "AllStatsAggregators"))
+local StatsButtonClass = require(folder:WaitForChild( 
+    "StatsButton"))
+local StatsViewerClass = require(folder:WaitForChild( 
+    "StatsViewer"))
+local StatsUtils = require(folder:WaitForChild( 
+    "StatsUtils"))
+
+--[[ Fast Flags ]]--
+local getShowPerformanceStatsInGuiSuccess, showPerformanceStatsInGuiValue = 
+	pcall(function() return settings():GetFFlag("ShowPerformanceStatsInGui") end)
+local showPerformanceStatsInGui = getShowPerformanceStatsInGuiSuccess and showPerformanceStatsInGuiValue
+
+
+--[[ Script Variables ]]--
+local screenGui = Instance.new("ScreenGui")
+screenGui.Name = "PS_Gui"
+local masterFrame = Instance.new("Frame")
+masterFrame.Name = "PS_MasterFrame"
+local localPlayer = PlayersService.LocalPlayer
+
+local allStatsAggregators = AllStatsAggregatorsClass.new()
+local statsViewer = StatsViewerClass.new()
+local statsButtonsByType ={}
+local currentDisplayType = nil
+
+for i, displayType in ipairs(StatsUtils.AllStatDisplayTypes) do
+  local button = StatsButtonClass.new(displayType)
+  statsButtonsByType[displayType] = button
+end
+
+--[[ Functions ]]--
+
+-- Set localPlayer to value of Players.localPlayer
+-- Update our screenGui parent accordingly.
+function UpdateLocalPlayer()
+  localPlayer = PlayersService.LocalPlayer
+  if localPlayer then
+    screenGui.Parent = localPlayer:WaitForChild("PlayerGui")
```

Comment:
specifically talking about screenGui.Parent. So we create this screenGui and then parent it to localPlayer.PlayerGui. PlayerGui is accessible by developer scripts, its also where they would put their own 2d UI for their game. We have a separate container for roblox created UI. Developers can't access this, and it renders on top of all developer UI. Looks like above you grab a reference to this container via CoreGuiService:WaitForChild("RobloxGui"). I'd parent this screenGui to that container instead.